el-patch.el

```diff
@@ -843,23 +845,26 @@ See `el-patch-validate'."
 
 ;;;;; el-patch-feature
 
+(defvar el-patch--features nil
+  "Alist of features to load before validating patches.
+Each entry in the alist is a list of arguments to pass to
+`el-patch-require-function', starting with the feature name as a
+symbol.")
+
 ;;;###autoload
 (defmacro el-patch-feature (feature &rest args)
-  "Declare that some patches are only defined after FEATURE is loaded.
-This is a convenience macro that creates a function for invoking
-`require' on that feature, and then adds it to
-`el-patch-pre-validate-hook' so that your patches are loaded and
-`el-patch' can properly validate them.
-
-FEATURE should be an unquoted symbol. ARGS, if given, are passed
-as quoted literals along with FEATURE to
-`el-patch-require-function' when `el-patch-validate-all' is
-called."
-  (let ((defun-name (intern (format "el-patch-require-%S" feature))))
-    `(progn
-       (defun ,defun-name ()
-         (apply el-patch-require-function ',feature ',args))
-       (add-hook 'el-patch-pre-validate-hook #',defun-name))))
+  "Declare that FEATURE is needed before patch validation.
+ARGS are quoted literals that will be passed along with FEATURE
```

Comment:
Small critique, this is not really correct. There is a big difference between "ARGS are passed as quoted literals" and "ARGS _are_ quoted literals". The latter suggests that the user should quote them while the former says that the function will quote them automatically.